temba/channels/handlers.py

```diff
@@ -207,6 +207,45 @@ def post(self, request, *args, **kwargs):
         return HttpResponse("Not Handled, unknown action", status=400)
 
 
+class TwimlAPIHandler(View):
+    @disable_middleware
+    def dispatch(self, *args, **kwargs):
+        return super(TwimlAPIHandler, self).dispatch(*args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        return self.post(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        from twilio.util import RequestValidator
+        from temba.msgs.models import Msg
+
+        signature = request.META.get('HTTP_X_TWILIO_SIGNATURE', '')
+        url = "https://" + settings.HOSTNAME + "%s" % request.get_full_path()
+
+        action = kwargs['action']
+        channel_uuid = kwargs['uuid']
+
+        channel = Channel.objects.filter(uuid=channel_uuid, is_active=True, channel_type=TWIML_API).exclude(org=None).first()
+        if not channel:
+            return HttpResponse("Channel with uuid: %s not found." % channel_uuid, status=404)
+
+        if action == 'receive':
+
+            org = channel.org
+            client = org.get_twiml_client()
+            validator = RequestValidator(client.auth[1])
+
+            if not validator.validate(url, request.POST, signature):
+                # raise an exception that things weren't properly signed
+                raise ValidationError("Invalid request signature")
+
+            Msg.create_incoming(channel, URN.from_tel(request.POST['From']), request.POST['Body'])
+
+            return HttpResponse("", status=201)
```

Comment:
Add a body here, we generally have something like "Msg Accepted: 15195"

Where is this handling incoming calls?